Need to set JAVA_HOME to build jphyloref

Hi there :wave: I'm Markus, I'm doing one of the reviews for JOSS. I'll open issues here as I go through the checklist at https://github.com/openjournals/joss-reviews/issues/3374

When trying to build `jphyloref` using `mvn package` on my machine (Ubuntu 20.04, openjdk 11.0.11) I get the following error:
```
<lots of lines with INFO and WARNING>
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  22.897 s
[INFO] Finished at: 2021-06-17T14:40:02+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (attach-javadocs) on project jphyloref: MavenReportException: Error while generating Javadoc: Unable to find javadoc command: The environment variable JAVA_HOME is not correctly set. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
Full log: [package.2.log](https://github.com/phyloref/jphyloref/files/6670243/package.2.log)

Infact, the error message `The environment variable JAVA_HOME is not correctly set.` was helpful in finding a solution. Simply running:
```
JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64 mvn package
```
worked for me. This might be worth mentioning in the installation instructions in the README.

--- a/README.md
+++ b/README.md
@@ -33,6 +33,11 @@ Development of JPhyloRef takes place in [our GitHub repository]. This includes [
 
 # Build and execution instructions
 
+You will need [Java] and [Apache Maven] to build the software from source. We recommend installing these
+tools using a package manager, such as [Homebrew] on macOS. You can use [jEnv] instead if you want to install
+multiple Java versions on the same computer. Installation should set up the `JAVA_HOME` environment
+variable; if not, you will need to set it to point at the directory containing your Java installation.
+
 Once you have downloaded the source code to your computer, you can compile and test the code by running `mvn test`.
 
 JPhyloRef can be built from source by running `mvn package` from the root directory of this repository. This will create a JAR file in the `target/` directory, which can be executed by running,
@@ -88,6 +93,10 @@ Once you're set up, you can run `mvn clean deploy` to publish the package
 to the OSSRH. If your version number ends in `-SNAPSHOT`, this will be
 published to the OSSRH Snapshots repository.
 
+  [Java]: https://www.java.com/en/
+  [Apache Maven]: https://maven.apache.org/
+  [Homebrew]: https://brew.sh/
+  [jEnv]: https://www.jenv.be/
   [Sonatype OSSRH]: https://central.sonatype.org/pages/ossrh-guide.html
   [the Sonatype website]: https://central.sonatype.org/pages/apache-maven.html
   [Maven settings.xml file]: https://central.sonatype.org/pages/apache-maven.html

--- a/README.md
+++ b/README.md
@@ -19,6 +19,8 @@ can be used to resolve [phyloreferences](http://phyloref.org):
     - You can also use the `/version` endpoint to test whether the software is
       working. It will report on the version of JPhyloRef, OWLAPI and reasoner
       being used.
+    - Note that no content is served at `/`; you will need to use `/version` to
+      test that the server is running.
 - `java -jar jphyloref.jar test input.owl`: Test all the phyloreferences in
   `input.owl` by comparing their resolution with the expected resolution recorded
   in the file.

--- a/README.md
+++ b/README.md
@@ -33,6 +33,8 @@ Development of JPhyloRef takes place in [our GitHub repository]. This includes [
 
 # Build and execution instructions
 
+Once you have downloaded the source code to your computer, you can compile and test the code by running `mvn test`.
+
 JPhyloRef can be built from source by running `mvn package` from the root directory of this repository. This will create a JAR file in the `target/` directory, which can be executed by running,
 for example:
 

--- a/pom.xml
+++ b/pom.xml
@@ -242,6 +242,16 @@
               </configuration>
             </plugin>
 
+            <!-- Recreate JAR file on every run -->
+            <plugin>
+              <groupId>org.apache.maven.plugins</groupId>
+	      <artifactId>maven-jar-plugin</artifactId>
+	      <version>3.2.0</version>
+	      <configuration>
+                <forceCreation>true</forceCreation>
+	      </configuration>
+	    </plugin>
+
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>

--- a/pom.xml
+++ b/pom.xml
@@ -97,13 +97,6 @@
             <version>2.3.2</version>
         </dependency>
 
-        <!-- https://mvnrepository.com/artifact/com.github.jsonld-java/jsonld-java -->
-        <dependency>
-            <groupId>com.github.jsonld-java</groupId>
-            <artifactId>jsonld-java</artifactId>
-            <version>0.12.0</version>
-        </dependency>
-
         <!-- REASONERS -->
 
         <!-- Elk is an OWL 2 EL reasoner -->

--- a/pom.xml
+++ b/pom.xml
@@ -108,10 +108,10 @@
         </dependency>
 
         <!-- To access OWL-API -->
-        <!-- https://mvnrepository.com/artifact/net.sourceforge.owlapi/owlapi-distribution -->
+        <!-- https://mvnrepository.com/artifact/net.sourceforge.owlapi/owlapi-apibinding -->
         <dependency>
             <groupId>net.sourceforge.owlapi</groupId>
-            <artifactId>owlapi-distribution</artifactId>
+            <artifactId>owlapi-apibinding</artifactId>
             <version>4.5.19</version>
         </dependency>
 

